### DISCUSSION TITLE
How does Cactus handle if a already accepted transaction has been rolled back

### DISCUSSION THREAD

**Discussion by brsbalazs** [hyperledger/cactus#704] [Q&A]
I am experimenting with the Cactus electricity trade example project, and asking for your help in connection the topic mentioned above. How does Cactus handle if both two ledgers connected with Cactus accept a transaction and the transaction is recorded in a block in both ledgers, but afterwards because of some reason the already accepted transaction is being rolled back on one of the ledgers? For example sticking to the model used in the electricity trade example project, if the transaction is being accepted on the monitored Sawtooth ledger, and the transaction is also written in a block on the Ethereum node, but then the block is being rejected by the main Ethereum node. Is the verifier component able to detect and handle this event? If yes, what is the scenario of rolling back a transaction like this? Thanks for your answer.

**Comment by petermetz** [comment]
@brsbalazs We have HTLCs in the works for atomic swaps of assets (balances) where the funds are returned after a certain period of time if the counter party did not satisfy their side of obligations. 
This (the above) is not a real rollback (more like an eventually reached correct state guarantee) because the ledgers that we support simply don't have a rollback feature so unfortunately this is the best one can do in this situation (in other words Cactus doesn't do any sort of magic)

Specifically about the electricity trade project: I don't know much about that one but I can cc here the maintainers who wrote it maybe they can help out. @sfuji822 @takeutak

**Reply by brsbalazs** [reply]
@petermetz Fine, many thanks for the detailed answer.
